src/Elm/Kernel/Platform.js

```diff
@@ -15,43 +15,177 @@ import Result exposing (isOk)
 // PROGRAMS
 
 
-var _Platform_worker = F4(function(impl, flagDecoder, debugMetadata, args)
+var _Platform_worker = F3(function(impl, flagDecoder, debugMetadata)
 {
-	return _Platform_initialize(
-		flagDecoder,
-		args,
-		impl.__$init,
-		impl.__$update,
-		impl.__$subscriptions,
-		function() { return function() {} }
-	);
+	var init = function(args)
+	{
+		return _Platform_initialize(
+			flagDecoder,
+			args,
+			null,
+			null,
+			null,
+			function() { return function() {} },
+			impl
+		);
+	};
+
+	/**__DEBUG/
+	init.hotReloadData = {
+		__$impl: impl,
+		__$platform_effectManagers: _Platform_effectManagers,
+		__$scheduler_enqueue: __Scheduler_enqueue
+	};
+	//*/
+
+	return init;
```

Comment:
All functions that wrap `_Platform_initialize` (also in elm/browser) have been changed this way:

- Decrease from `F4` to `F3` and move the last argument to an explicit function (called `init`) inside. That’s equivalent – just another way to express the curriedness.
- Attach `.hotReloadData` on that inner function in `__DEBUG` mode.

The inner function is called `init` because that’s the `init` function that ends up at `Elm.Main.init` for example. So for each Elm entrypoint, there’s `.init()` to initialize an app as usual, and `.init.hotReloadData` for use when hot reloading already initialized apps of that entrypoint.

Code that I’ll comment on later removes `.hotReloadData`, so there’s no risk that people start depending on it.